contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -4796,3 +4802,33 @@ pltsql_revert_guc(int nest_level)
        PltsqlGUCNestLevel = nest_level - 1;
 
 }
+
+void
+check_current_query_is_create_tbl_check_constraint(Node *expr)
+{
+	CreateStmt *stmt = (CreateStmt *) expr;
+	ListCell   *elements;
+	bool		contains_check_constraint = false;
+
+	foreach(elements, stmt->tableElts)
+	{
+		Node *element = lfirst(elements);
+		switch (nodeTag(element))
+		{
+			case T_Constraint:
+			{
+				Constraint *c = (Constraint *) element;
+				if(c->contype == CONSTR_CHECK)
+					contains_check_constraint = true;
+			}
+				break;
+			default:
+				break;
+		}
+		if (contains_check_constraint)
+		{
+			current_query_is_create_tbl_check_constraint = true;
+			break;
+		}
+	}
```

Comment:
The code is functional, but I'd suggest to polish a little bit so that it could be more succinct.